functions/controllers/announcements-api/index.js

```diff
@@ -0,0 +1,68 @@
+const functions = require("firebase-functions");
+const { db } = require("../../utils/admin");
+const helmet = require("helmet");
+const express = require("express");
+const cors = require("cors");
+const { corsConfig } = require("../../utils/config");
+const { hasPermission } = require("../../utils/middleware");
+const { addDocument, queryDocument } = require("../../utils/database");
+
+const announcements = express();
+const corsOptions = {
+  origin: corsConfig,
+  optionsSuccessStatus: 200,
+};
+
+announcements.disable("x-powered-by");
+announcements.use(helmet());
+announcements.use(cors(corsOptions));
+announcements.use(express.json());
+
+
+// Read all
+announcements.get("/", async (req, res) => {
+  // orderBy() should default to ascending, double check
+  const snapshot = await db.collection("announcements").orderBy("timeStamp").get();
+  if (snapshot.empty) {
+    return res.status(200).send(JSON.stringify([]));
```

Comment:
Any API we implement, we want to have a clear response schema. So we should never just res.send(data), it should almost always have a message associated with the response such as "announcements retrieved" or "Ok" to indicate to a client with the data as a separate field.